src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] dataElements;
+    private int size = 0;
+
+    public ArrayList() {
+        dataElements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == dataElements.length) {
+            grow();
+        }
+        dataElements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+
+        if (size == dataElements.length) {
+            grow();
+        }
 
+        System.arraycopy(dataElements, index, dataElements, index + 1, size - index);
+        dataElements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null || list.isEmpty()) {
+            return;
+        }
 
+        ensureCapacity(size + list.size());
+
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
```

Comment:
The 'add' method already contains logic for growing the array. It's better to use 'add(value, index)' method here to avoid array resizing in each iteration. It will improve the performance of your code.